src/main/java/approximations/java/SelfTestMain.java

```diff
@@ -0,0 +1,305 @@
+package approximations.java;
```

Comment:
You are using wrong version of the "self-testing" class.
Consider removing this implementation and switching to using `approximations.SelfTestMain` class.